### PR TITLE
doc: Unit test documentation for nRF9160 Asset Tracker v2

### PR DESCRIPTION
**Pull request by divipillai**
Add unit test documentation for nRF9160 Asset Tracker v2

Signed-off-by: divya pillai <divya.pillai@nordicsemi.no>

### CODE REVIEW

File: applications/asset_tracker_v2/doc/unit_test.rst [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+Unit Tests
+###########
+
+.. contents::
+   :local:
+   :depth: 2
+
+The internal modules of :ref:`asset_tracker_v2` have a set of unit tests in the :file:`asset_tracker_v2/tests` folder.
+Following are the modules which contains unit tests:
```
Comment (coderbyheart) [inline applications/asset_tracker_v2/doc/unit_test.rst:11]:
```suggestion
Following are the modules which contain unit tests:
```

**Review by coderbyheart — CHANGES_REQUESTED** [review]
Typo in commit message.

**Review by coderbyheart — APPROVED** [review]

File: applications/asset_tracker_v2/doc/unit_test.rst [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+.. code-block:: console
+
+   twister -T
+   
+* To execute the tests on the ``qemu_cortex_m3`` board targets:
+
+.. code-block:: console
+
+   west build -b qemu_cortex_m3 -t run
```
Comment (balaji-nordic) [inline applications/asset_tracker_v2/doc/unit_test.rst:53]:
```suggestion
   twister -T -p qemu_cortex_m3
```

File: applications/asset_tracker_v2/doc/unit_test.rst [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+Run unit tests using Twister
+============================
+
+* To execute the tests on both :ref:`zephyr:native_posix` and ``qemu_cortex_m3`` board targets:
+
+.. code-block:: console
+
+   twister -T
+   
+* To execute the tests on the ``qemu_cortex_m3`` board targets:
```
Comment (balaji-nordic) [inline applications/asset_tracker_v2/doc/unit_test.rst:49]:
```suggestion
* To execute the tests on the ``qemu_cortex_m3`` board target:
```

File: applications/asset_tracker_v2/doc/unit_test.rst [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+   west build -b qemu_cortex_m3 -t run
+
+Run unit tests using Twister
+============================
+
+* To execute the tests on both :ref:`zephyr:native_posix` and ``qemu_cortex_m3`` board targets:
+
+.. code-block:: console
+
+   twister -T
```
Comment (balaji-nordic) [inline applications/asset_tracker_v2/doc/unit_test.rst:47]:
```suggestion
   twister -T .   
```

The -T option should be followed by the directory where the tests are.

File: applications/asset_tracker_v2/doc/unit_test.rst [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+   twister -T . -p nrf9160dk_nrf9160_ns --device-testing --device-serial <serial port>
+
+In this console snippet:
+ 
+* ``device-serial`` - This parameter is mandatory for Twister to work.
+* ``serial port`` - This option must point to the port where you receive logs from the DK, normally, the first port listed by ``nrfjprog --com``. 
+  For example, ``/dev/ttyACM0``.
+
+The :file:`testcase.yaml` file for that unit test must have the entry ``platform_allow: nrf9160dk_nrf9160_ns``.
+See :file:`nrf/applications/asset_tracker_v2/tests/gps_module/testcase.yaml` for an example.
```
Comment (balaji-nordic) [inline applications/asset_tracker_v2/doc/unit_test.rst:69]:
```suggestion
See :file:`nrf/applications/asset_tracker_v2/tests/gnss_module/testcase.yaml` for an example. 
```

**Review by balaji-nordic — CHANGES_REQUESTED** [review]
Thanks for this. Requested a few changes.

@shantha-14  This is FYI. Here we are adding documentation on how to run the asset tracker v2 unit tests.

File: applications/asset_tracker_v2/README.rst [hunk 6]
```diff
@@ -29,3 +29,4 @@ See the subpages for detailed documentation on the application and its internal
    doc/ui_module
    doc/util_module
    doc/modules_common
+   doc/unit_test
```
Comment (umapraseeda) [inline applications/asset_tracker_v2/README.rst:32]: Not sure if there is a spacing issue here. Please check.
```suggestion
  doc/ui_module
  doc/util_module
  doc/modules_common
  doc/unit_test
```

Comment (divipillai) [reply applications/asset_tracker_v2/README.rst:32]: There was no spacing issue. It renders correctly. Not sure why git is showing the spacing issue.

File: applications/asset_tracker_v2/doc/unit_test.rst [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+.. code-block:: console
+
+   twister -T . -p nrf9160dk_nrf9160_ns --device-testing --device-serial <serial port>
+
+In this console snippet:
+
+* ``device-serial`` - This parameter is mandatory for Twister to work.
+* ``serial port`` - This option must point to the port where you receive logs from the DK, normally, the first port listed by ``nrfjprog --com``.
+  For example, ``/dev/ttyACM0``.
```
[the remark this answers is not in the document]
Comment (balaji-nordic) [reply applications/asset_tracker_v2/doc/unit_test.rst:72]: The statement "``device-serial`` - This parameter is mandatory for Twister to work." is a bit unnecessary.

**Review by balaji-nordic — APPROVED** [review]
Consider joining line 68 and line 69 in unit_test.rst. I think line 69 does not need to be in a new line. Everything else looks good.

**Review by simensrostad — APPROVED** [review]

File: applications/asset_tracker_v2/doc/unit_test.rst [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+* ``qemu_cortex_m3`` board target:
+
+.. code-block:: console
+
+   west build -b qemu_cortex_m3 -t run
+
+Running unit tests using Twister
+================================
+
+Enter the following twister command to execute the tests on different board targets:
```
Comment (umapraseeda) [inline applications/asset_tracker_v2/doc/unit_test.rst:45]:
```suggestion
Enter the following twister commands to execute the tests on different board targets:
```

**Review by umapraseeda — APPROVED** [review]

File: applications/asset_tracker_v2/doc/unit_test.rst [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+To run the unit test, you must navigate to the test directory of the respective internal module.
+For example, to run the unit test for :ref:`asset_tracker_v2_debug_module`, navigate to :file:`asset_tracker_v2/tests/debug_module`.
+The unit tests can be executed using West or Twister.
+
+Running unit tests using West
+=============================
+
+Enter the following west commands to execute the tests on different board targets:
+
+* :ref:`zephyr:native_posix` board target:
```
Comment (Dummy1134) [inline applications/asset_tracker_v2/doc/unit_test.rst:30]: - [ ] []()@NordicBuilder 
```suggestion
* :ref:`zephyr:native_posix` board target: 🎯

```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 10]
```diff
@@ -103,6 +103,7 @@ nRF9160: Asset Tracker v2
   * Support for QEMU x86 emulation.
   * Support for the :ref:`lib_nrf_cloud_pgps` flash memory partition under certain conditions.
```
Comment (Dummy1134) [inline doc/nrf/releases/release-notes-changelog.rst:104]: Run
applications/asset_tracker_v2/doc/unit_test.rstapplications/asset_tracker_v2/doc/unit_test.rst

**Review by Dummy1134 — APPROVED** [review]

**Pull request merged**